functions/src/modules/gmail.functions.ts

```diff
@@ -0,0 +1,273 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import * as Google from 'googleapis';
+import * as querystring from 'querystring';
+import { Message } from 'firebase-functions/lib/providers/pubsub';
+
+const db = admin.database();
+
+export const oauth2init = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    const { client_key, secret, redirect } = functions.config().gmail;
+    const oauth2Client = new Google.google.auth.OAuth2(
+      client_key,
+      secret,
+      redirect
```

Comment:
“Call back should be explicitly specified” means that we should construct the callback in the code, not specifying it in the config. Is it possible? My concern was that in the article they don't have function namespaces, but we have. And they assumed that `oauth2callback` is a callback path by default… Anyhow please shed light on this.